frontend/src/components/editor/Preview.tsx

```diff
@@ -1,23 +1,50 @@
 import { CircularProgress, Stack } from "@mui/material";
-import MarkdownPreview from "@uiw/react-markdown-preview";
-import katex from "katex";
 import "katex/dist/katex.min.css";
-import { useEffect, useState } from "react";
+import { useEffect, useRef, useState } from "react";
 import { useSelector } from "react-redux";
-import rehypeExternalLinks from "rehype-external-links";
-import rehypeKatex from "rehype-katex";
-import { getCodeString } from "rehype-rewrite";
-import rehypeSanitize, { defaultSchema } from "rehype-sanitize";
-import remarkMath from "remark-math";
 import { useCurrentTheme } from "../../hooks/useCurrentTheme";
 import { selectEditor } from "../../store/editorSlice";
 import { addSoftLineBreak } from "../../utils/document";
+import MarkdownIt from "markdown-it";
+import { toHtml } from "hast-util-to-html";
+import markdownItKatex from "@vscode/markdown-it-katex";
+import { refractor } from "refractor";
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
+import markdownItIncrementalDOM from "markdown-it-incremental-dom";
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
+import markdownItSanitizer from "markdown-it-sanitizer";
+// eslint-disable-next-line
+const IncrementalDOM = require("incremental-dom");
```

Comment:
I used required() because of [this issue](https://github.com/google/incremental-dom/issues/440)